### ISSUE TITLE
Create api reference category descriptions for web APIs

### ISSUE THREAD

**Issue by donjo** [denoland/deno-docs#487]
Descriptions for the web API categories that would show up on an index page like this.

![Image](https://github.com/denoland/deno-docs/assets/776987/34c7bd2c-1f89-42f7-acd1-67f37a4be0bc)

**Comment by thisisjofrank** [comment]
**FFI**
Foreign Function Interface. Call functions from shared libraries (e.g., C/C++) directly from Deno.
Useful for integrating with existing native code or accessing low-level system functionality.

**I/O**
Input/Output.  Reading and writing files, working with streams, and handling standard input/output.

**Permissions**
Permission system controls access to resources (e.g., file system, network, environment variables).
Request permissions explicitly, enhancing security and user trust.

**Jupyter**
Create interactive notebooks and execute code cells. Useful for data analysis, visualization, and educational purposes.

**Cloud**
Tools for managing state, scheduling tasks, and interacting with key-value stores.

**File System**
APIs for working with files, directories, and file metadata. Includes functions for reading, writing, and manipulating file paths.

**Network**
A wide range of networking tasks, from low-level connections to high-level server creation. Handle HTTP requests, WebSocket communication, and DNS resolution. Useful for building web servers, clients, and networking tools. 

**HTTP Server**
Handling HTTP requests, serving responses, and managing server behavior.

**Sub Process**
Spawn and manage child processes, execute commands, and collect output. Useful for executing external programs from Deno.

**Errors**
Error types and utilities for handling exceptions and custom error scenarios. Helps improve error handling and debugging.

**Runtime Env**
System-related functionality, process management, and observability.

**Testing & Benchmarking**
 Robust testing and benchmarking capabilities to ensure code quality and performance.


@crowlKats @ry how do these look?

**Comment by lucacasonato** [comment]
> I/O
Input/Output. Reading and writing files, working with streams, and handling standard input/output.

Reading and writing files is in `File system`, no?

**Comment by lucacasonato** [comment]
> Network
> A wide range of networking tasks, from low-level connections to high-level server creation. Handle HTTP requests, WebSocket communication, and DNS resolution. Useful for building web servers, clients, and networking tools.
>
> HTTP Server
> Handling HTTP requests, serving responses, and managing server behavior.

Not clear how these are different. If `Network` contains "Handle HTTP requests", then what does `Http Server` contain?

**Comment by lucacasonato** [comment]
> Runtime Env
> System-related functionality, process management, and observability.

Should explicitly mention environment variables.

**Comment by thisisjofrank** [comment]
> > Network
> > A wide range of networking tasks, from low-level connections to high-level server creation. Handle HTTP requests, WebSocket communication, and DNS resolution. Useful for building web servers, clients, and networking tools.
> > HTTP Server
> > Handling HTTP requests, serving responses, and managing server behavior.
> 
> Not clear how these are different. If `Network` contains "Handle HTTP requests", then what does `Http Server` contain?

yeah, when i was writing this i was struggling to remember why we separated these categories...

**Comment by thisisjofrank** [comment]
> > I/O
> > Input/Output. Reading and writing files, working with streams, and handling standard input/output.
> 
> Reading and writing files is in `File system`, no?

have updated

**Comment by thisisjofrank** [comment]
*Web APIs*
A set of essential interfaces and functions to interact with the browser environment. Eg handling user interactions, getting browser and device information, scheduling and canceling periodic or one-time tasks.


*Canvas APIs*
Create, transform, and display images and graphics within the HTML <canvas> element.

*URL*
Manipulate URLs, extract data from URLs and manage query parameters.

*Events*
Handle events and interactions. Listen for and respond to various events, including custom events, errors, progress, and promise rejections.

*Fetch APIs*
Tools to fetch data. Retrieve resources from servers, handle responses, and manage network requests.

*Cache APIs*
Tools for caching and managing data. Store and retrieve data from the cache, manage cache storage limits, and handle cache-related events.

*Messaging APIs*
Facilitate communication between different parts of an application, allowing data exchange and coordination.

*Encoding APIs*
Handle character encoding, decoding, and binary data conversion.

*Intl*
Tools for internationalization and localization. Create language-aware applications, handle date and time formatting, and adapt content for different locales.

*Performance*
Measure, analyze, and optimize application performance.

*Streams APIs*
Manage data streams, queuing strategies, and transformations. Handle data in chunks, process large datasets, and optimize memory usage.

*WebAssembly*
Efficiently execute computationally intensive tasks. WASM module compilation, instantiation, memory management, and interaction with imports and exports.

*Web File APIs*
File and data manipulation tools. Handle files, read data and work with binary content.

*WebSockets*
Enable real-time communication between clients and servers using WebSockets. Tools to create interactive and dynamic applications.

*Web Storage APIs*
Store data locally within the browser. Manage session storage and local storage.

*Web Workers*
Run script operations in background threads. Manage worker threads, communicate with workers, and handle data transfer between workers and the main thread.

*Web Crypo APIs*
Cryptographic functionality for secure communication, data protection, and key management. Generate secure random numbers, encrypt and decrypt data, and manage cryptographic keys.

*Temporal APIs*
Date and time handling. Includes long-lived Workflows, calendar systems, time zones, and precise duration calculations.

*WebGPU APIs*
GPU programming and rendering. Efficiently use a device’s graphics processing unit (GPU) for high-performance computations and complex image rendering.

**Comment by ry** [comment]
> Fetch APIs
> Tools to fetch data. Retrieve resources from servers, handle responses, and manage network requests.

HTTP client 

> Messaging APIs
> Facilitate communication between different parts of an application, allowing data exchange and coordination.


Not sure what this would include
